### ISSUE TITLE
Way to indicate out of range?

### ISSUE THREAD

**Issue by chmorgan** [sgreben/piecewiselinear#2]
Would a pull request to modify At() to return (float64, error) be accepted? In my use case I'd prefer not to rely on a magic value of 0 to indicate not found but would propose instead we returned something like 0, fmt.Errorf("value out of range") so the user could decide how to handle those cases?

**Comment by sgreben** [comment]
Hey, thanks for getting in touch! Personally I'd prefer keeping the signature for `At` as-is. Open to discussion though :) Would something like `func (f Function) IsInterpolatedAt(x float64) bool` perhaps also cover your use-case? So you could then do

```
if !f.IsInterpolatedAt(x) {
   // x is not within domain X, f.At(x) is guaranteed to be zero
   return
}
//  x is within domain X, f.At(x) might be non-zero
```

I actually find it useful that the function is well-defined outside the X domain - this way it can be used wherever a `func (float)float` is expected.

**Comment by chmorgan** [comment]
That would work for me. Do you want me to submit a pr to do that with tests?

Chris

**Comment by sgreben** [comment]
Yes, please! And if you have a better idea for the name than I did (`IsInterpolatedAt` doesn't exactly roll off the tongue...) please feel free to choose that :) Many thanks!

Sergey

**Comment by chmorgan** [comment]
Hi @sgreben I opened a PR and commented about the implementation. I wasn't sure there was a clean way to figure out if the value was out of range so went the route of extending the existing At() function. I would appreciate feedback and can adjust if necessary.

**Comment by sgreben** [comment]
Addressed in #3 Thanks!
